distributed/scheduler.py

```diff
@@ -4573,7 +4603,21 @@ def update_graph(
         # Compute recommendations
         recommendations: dict = {}
 
-        for ts in sorted(runnables, key=operator.attrgetter("priority"), reverse=True):
+        sorted_tasks = sorted(
+            runnables, key=operator.attrgetter("priority"), reverse=True
+        )
+
+        if self.cogroups:
+            start = max(self.cogroups.keys()) + 1
+        else:
+            start = 0
+        cogroups = coassignmnet_groups(sorted_tasks[::-1], start=start)
+        self.cogroups.update(cogroups)
+        for gr_ix, tss in self.cogroups.items():
+            for ts in tss:
+                ts.cogroup = gr_ix
```

Comment:
If one wants to go with a plain `dict` for maintaining cogroups, I think it would make more sense if this invariant were maintained in `coassignment_groups` (see below).

Reply:
I haven't decided, yet, what to use to maintain this. Maintenance of this structure is not implemented yet (e.g. we're not cleaning it up again). For now, I am using a dict for simplicity. I'm also not set on gr_ix being an integer fwiw